src/dbt/kipptaf/models/performance_management/intermediate/int_performance_management__overall_scores.sql

```diff
@@ -0,0 +1,165 @@
+    all_scores as (
+        select
+            employee_number,
+            academic_year,
+            pm_term,
+            etr_score,
+            so_score,
+            overall_score,
+            etr_tier,
+            so_tier,
+            overall_tier,
+            case
+                when pm_term = 'PM1'
+                then date(academic_year, 10, 1)
+                when pm_term = 'PM2'
+                then date(academic_year + 1, 1, 1)
+                when pm_term = 'PM3'
+                then date(academic_year + 1, 3, 1)
+                when pm_term = 'PM4'
+                then date(academic_year + 1, 5, 15)
+            end as eval_date,
+        from
+            {{
+                source(
+                    "performance_management",
+                    "src_performance_management__scores_overall_archive",
+                )
+            }}
+
+        union all
+
+        select distinct
+            employee_number,
+            academic_year,
+            form_term,
+            etr_score,
+            so_score,
+            overall_score,
+            etr_tier,
+            so_tier,
+            overall_tier,
+            case
+                when form_term = 'PM1'
+                then date(academic_year, 10, 1)
+                when form_term = 'PM2'
+                then date(academic_year + 1, 1, 1)
+                when form_term = 'PM3'
+                then date(academic_year + 1, 3, 1)
+            end as eval_date,
+        from {{ ref("rpt_tableau__schoolmint_grow_observation_details") }}
+        where form_type = 'PM' and rn_submission = 1
+
+        union all
+
+        select
+            employee_number,
+            academic_year,
+            'PM4' as form_term,
+            avg(etr_score) as etr_score,
+            avg(so_score) as so_score,
+            avg(overall_score) as overall_score,
+            case
+                when avg(etr_score) >= 3.5
+                then 4
+                when avg(etr_score) >= 2.745
+                then 3
+                when avg(etr_score) >= 1.745
+                then 2
+                when avg(etr_score) < 1.75
+                then 1
+            end as etr_tier,
+            case
+                when avg(so_score) >= 3.5
+                then 4
+                when avg(so_score) >= 2.945
+                then 3
+                when avg(so_score) >= 1.945
+                then 2
+                when avg(so_score) < 1.95
+                then 1
+            end as so_tier,
+            case
+                when avg(overall_score) >= 3.5
+                then 4
+                when avg(overall_score) >= 2.745
+                then 3
+                when avg(overall_score) >= 1.745
+                then 2
+                when avg(overall_score) < 1.75
+                then 1
+            end as overall_tier,
+            date(academic_year + 1, 5, 15) as eval_date,
+        from {{ ref("rpt_tableau__schoolmint_grow_observation_details") }}

```

Comment:
circular dependency, we need to refactor the upstream views to make this work in one direction

Reply:
wrong ref